sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/ControlFlowGraph/CSharpControlFlowGraphBuilder.cs

```diff
@@ -558,25 +558,49 @@ private Block BuildTryStatement(TryStatementSyntax tryStatement, Block currentBl
                 tryEndStatementConnections.Add(this.ExitTarget.Peek());
             }
 
-            var tryBody = BuildBlock(tryStatement.Block,
-                CreateBranchBlock(tryStatement, tryEndStatementConnections.Distinct()));
-
-            var tryStartStatementConnections = catchBlocks.ToList();
-            tryStartStatementConnections.Add(tryBody); // try body
-
-            if (!areAllExceptionsCaught) // unexpected exception thrown, go to exit (through finally if present)

```

Comment:
the `areAllExceptionsCaught` variable is still used at line 556 where it is used for the `tryEndStatementConnections`.

The main change of this PR is that I removed from the `beforeTry` block (initially called `tryStartBlock`) the edges to the `catch` and `finally` blocks - what happens outside of a `try` block (and `tryStartBlock` is outside the try block) won't be caught by `catch` and won't enter `finally` !